### ISSUE TITLE
Improve indicator search in the chart editor

### ISSUE THREAD

**Issue by Marigold** [owid/owid-grapher#4581]
Some ideas for improving the indicator search in the **chart editor**. This was initially suggested by @spoonerf on Slack as one of her [pain points](https://owid.slack.com/archives/C07S865NMRT/p1732798117078579). I’m documenting it here as an issue due to my work on indicator semantic search.

> It can sometimes be tricky to find the exact indicator you are looking for in the chart editor tool. It’s really useful to be able to search using exact terms (e.g.“age-standardized”), to exclude specific terms (e.g. -age-standardized), and to be able to filter by namespace.
Could we also add filtering by short_name and version too? This would make it a lot easier to narrow down which variable you want to add.

> For example, I recently went through a number of charts like this, switching them from age-standardized rates to crude rates, and it’s just currently quite fiddly to find the indicator you want, e.g., in the video here at first, five different datasets show up with the simpler search term. Yes, you can filter quite quickly using the tricks mentioned above, but that gets a little tiring when you need to do it multiple times.
I think at least making it easier to select a single dataset you want to search within would help.

To summarize:

- It would be helpful if the chart editor search had the same features as the [indicator search in admin](https://admin.owid.io/admin/variables).
- Add filtering by `shortName`, or possibly even `catalogPath`.
- Make it easier to filter by a single dataset, potentially using `catalogPath`.
- Marcel suggested searching across all of these: indicator name, dataset name, and `catalogPath`.

Semantic search isn’t as helpful in this context. However, if we start using the "Indicator search" app in Wizard, it would be straightforward to add these filtering features there for testing and later incorporate them into the chart editor's indicator search.

**Comment by ikesau** [comment]
https://github.com/owid/owid-grapher/issues/6014
